### PR TITLE
Added python postprocessing files for 2D-Backward-facing-step example

### PR DESCRIPTION
**Pull request by charles-lp**
This PR adds 3 files to the 2D-backward-facing-step example in order to let the users reproduce the analysis described in the example on their own.

The following files have been added : 

- **Bisection.py** : Computes the reattachment length with a bisection algorithm for a specified number of meshes
- **velocity_distribution.py** :  Computes the velocity distribution at inlet and outlet and compares it with analytical solution (Poiseuille)
- **benchmark_data.txt** : Contains the benchmark data used in the example to validate the reattachment length

### CODE REVIEW

**Review by blaisb — APPROVED** [review]
Nice, that part is mostly there to keep track of how you post-processed and made the results. Thanks for adding it :)

File: examples/incompressible_flow/2d-backward-facing-step/Bisection.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+file = list(range(0,n))
+data = list(range(0,n))
+N_elements = np.zeros(n)
+for i in range(0,n):
+    if i<10:
+        file[i] = ('Reynolds100-600/backward_facing_step_output.000' 
+                   + str(i) + '.0000.vtu')
+    elif i>=10:
+        file[i] = ('Reynolds100-600/backward_facing_step_output.00' 
+                   + str(i) + '.0000.vtu')
```
Comment (CaroleAnneDaunais) [inline examples/incompressible_flow/2d-backward-facing-step/Bisection.py:40]: In python you can use formating for this, such as : `f'{i:05d}'` (this is just a slight suggestion)

File: examples/incompressible_flow/2d-backward-facing-step/Bisection.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    
+# CONVERGENCE GRAPH
+mesh= np.linspace(0, n-1, n)
+plt.figure()
+plt.rc('axes', labelsize=15) 
+plt.scatter(mesh[1:], x_r[1:])
+plt.grid()
+plt.xticks([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
+plt.xlabel("Mesh refinement step")
+plt.ylabel(r'$x_r$/h')
```
Comment (CaroleAnneDaunais) [inline examples/incompressible_flow/2d-backward-facing-step/Bisection.py:85]: Maybe there could be a plot title, and do we want the plot to be saved in png format? The same comment goes for the next figure.

**Review by CaroleAnneDaunais — APPROVED** [review]
Apart from the two comments (that apply for the two .py files), it's very good! Nice PR :)

**Comment by charles-lp** [comment]
The requested changes has been added.

File: examples/incompressible_flow/2d-backward-facing-step/Bisection.py (hunk 1 above)
Comment (charles-lp) [reply examples/incompressible_flow/2d-backward-facing-step/Bisection.py]: Thank you for the info, this solved a LOT of pain in some of my scripts :)

**Review by blaisb — APPROVED** [review]

File: examples/incompressible_flow/2d-backward-facing-step/Bisection.py (hunk 1 above)
Comment (blaisb) [reply examples/incompressible_flow/2d-backward-facing-step/Bisection.py]: :). I learnt something today also ^^!. Lovely

**Pull request merged**